zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -329,7 +329,14 @@ cdef _array_for_column_impl(object dtype,
                 # upper bound doesn't include the timestamp because we've
                 # already included the timestamp-date in the baseline.
                 end = max(ts_ix - 1, 0)
-                if end >= asof_ix:
+                if end >= asof_ix and out_array[asof_ix, column_ix] != value:
+                    # The second condition of this ``and`` is technically
```

Comment:
This doesn't explain to a reader why we're skipping these events in the first place.